### PR TITLE
fix: skip controller test when no cap

### PR DESCRIPTION
**Pull request by guilhem**
Don't test controller calls when plugin don't advertise controller capability.

Signed-off-by: Guilhem Lettron <guilhem@barpilot.io>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/sanity/controller.go [hunk 1]
```diff
@@ -106,6 +106,41 @@ func isControllerCapabilitySupported(
 }
 
 var _ = DescribeSanity("Controller Service [Controller Server]", func(sc *TestContext) {
+	BeforeEach(func() {
+		ids := csi.NewIdentityClient(sc.Conn)
+		res, err := ids.GetPluginCapabilities(context.Background(), &csi.GetPluginCapabilitiesRequest{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).NotTo(BeNil())
+
+		controlled := false
```
Comment (pohly) [inline pkg/sanity/controller.go:115]: This seems like an odd name. Perhaps `isController` instead?

File: pkg/sanity/controller.go [hunk 2]
```diff
@@ -106,6 +106,41 @@ func isControllerCapabilitySupported(
...
+				default:
+					Fail(fmt.Sprintf("Unknown volume expansion mode: %v\n", cap.GetVolumeExpansion().GetType()))
+				}
+			default:
+				Fail(fmt.Sprintf("Unknown capability: %v\n", cap.GetType()))
+			}
+		}
+
+		if !controlled {
+			Skip("not controlled")
```
Comment (pohly) [inline pkg/sanity/controller.go:140]:
```suggestion
			Skip("does not implement controller")
```

**Comment by pohly** [comment]
@guilhem: are you still interested in finishing this PR?

**Comment by guilhem** [comment]
> @guilhem: are you still interested in finishing this PR?

Thanks for the ping @pohly :)
I will work on it right now.

**Comment by guilhem** [comment]
Last commit can be squashed :)

**Review by pohly — COMMENTED** [review]
Looks good to me, please squash.

**Comment by guilhem** [comment]
> Looks good to me, please squash.

done :)

**Comment by pohly** [comment]
After looking at this again, I started to wonder whether this current change goes far enough. There are other controller calls, for example in https://github.com/kubernetes-csi/csi-test/blob/master/pkg/sanity/node.go#L69.

If we merge this, can you run all tests?

**Pull request closed without merging**